### PR TITLE
Improve BenchmarkPartitionedOutputOperator

### PR DESCRIPTION
**Pull request by lukasz-stec**
## Description
Introduce a consistent TestType naming convention.
Add descriptions to the TestType cases.
Add more Dictionary test cases.
Add RowType with RLE field case.

Extracted (except for the docs commit) from https://github.com/trinodb/trino/pull/11289.

> Is this change a fix, improvement, new feature, refactoring, or other?

benchmark refactoring + documentation
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

only BenchmarkPartitionedOutputOperator
> How would you describe this change to a non-technical end user or system administrator?

Improved code readibility
## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java [hunk 1]
```diff
@@ -207,7 +212,7 @@ public Page createPage(List<Type> types, int positionCount, float nullRate)
                             createRandomLongsBlock(positionCount, 2));
                 }
             },
-            DICTIONARY_BIGINT(BigintType.BIGINT, 3000) {
+            DICTIONARY_BIGINT(BigintType.BIGINT, 5000) {
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java:247]: nit: that should be separate commit

File: core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java [hunk 2]
```diff
@@ -318,10 +323,45 @@ public OptionalInt getNullChannel()
...
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
+                }
+            },
             VARCHAR(VarcharType.VARCHAR, 5000),
+            DICTIONARY_VARCHAR(VarcharType.VARCHAR, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java:392]: nit: it would be better to use composition instead of overriding `createPage`. Right now you pass some arguments to `TestType`, but you also optionally override a method. With composition, you could reuse `createRandomDictionaryPage` rather than override it every time. Composition is also cleaner and more consistent than inheritance

File: core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java [hunk 3]
```diff
@@ -612,6 +622,14 @@ private static void pollute()
                 data.setupData(null);
                 data.setPageCount(1);
                 benchmark.addPage(data);
+                // pollute row-wise processing
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java:657]: this should go to commit that adds row-wise benchmark. BTW: do you know if it works?

**Review by sopel39 — APPROVED** [review]
lgtm % nits (optional)

File: core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java (hunk 2 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java]: makes sense. I refactored it to use new `PageGenerator` interface

File: core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java (hunk 3 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java:563]: there is no explicit row-wise benchmark. It has to be set up manually by choosing `positionCount` ~= `partitionCount`.

And yes, row-wise code path pollution works, it has visible influence on benchmark results:
```
### with row-wise pollution
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)  (type)  Mode  Cnt    Score    Error  U
BenchmarkPartitionedOutputOperator.addPage               1                false           0               512              512  BIGINT  avgt   10  167.598 ± 12.291  ms/op

### without row-wise pollution
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)  (type)  Mode  Cnt    Score    Error  Units
BenchmarkPartitionedOutputOperator.addPage               2                false           0               512              512  BIGINT  avgt   10  213.097 ± 14.498  ms/op
```

**Review by lukasz-stec — COMMENTED** [review]
as suggested I refactored the benchmark TestType to use page generation via composition over overriding the `createPage` method

**Review by sopel39 — APPROVED** [review]
